server.js

```diff
@@ -563,38 +569,41 @@ camp.route(/^\/sonar\/(http|https)\/(.*)\/(.*)\/(.*)\.(svg|png|gif|jpg|json)$/,
         try {
           var data = JSON.parse(buffer);
 
-          var percentage = data[0].msr[0].val;
+          var value = data[0].msr[0].val;
 
-          if (percentage === undefined) {
+          if (value === undefined) {
             badgeData.text[1] = 'unknown';
             sendBadge(format, badgeData);
             return;
           }
 
-          if (metric === 'tech_debt') {
+          if (metric === 'lines') {
+            badgeData.text[1] = value;
+            badgeData.colorscheme = 'green';
+          } else if (metric === 'tech_debt') {
             // colors are based on sonarqube default rating grid and display colors
             // [0,0.1)   ==> A (green)
             // [0.1,0.2) ==> B (yellowgreen)
             // [0.2,0.5) ==> C (yellow)
             // [0.5,1)   ==> D (orange)
             // [1,)      ==> E (red)
-            badgeData.text[1] = percentage + '%';
-            if (percentage > 100) {
+            badgeData.text[1] = value + '%';
+            if (value > 100) {
               badgeData.colorscheme = 'red';
-            } else if (percentage > 50) {
+            } else if (value > 50) {
               badgeData.colorscheme = 'orange';
-            } else if (percentage > 20) {
+            } else if (value > 20) {
               badgeData.colorscheme = 'yellow';
-            } else if (percentage > 10) {
+            } else if (value > 10) {
               badgeData.colorscheme = 'yellowgreen';
-            } else if (percentage > 0) {
+            } else if (value > 0) {
               badgeData.colorscheme = 'green';
             } else {
               badgeData.colorscheme = 'brightgreen';
             }
           } else {
-            badgeData.text[1] = percentage.toFixed(0) + '%';
-            badgeData.colorscheme = coveragePercentageColor(percentage);
+            badgeData.text[1] = value.toFixed(0) + '%';
```

Comment:
If the default condition is changed to just show the value with no percentage, another condition must be added to handle the metric 'coverage'.  Ideally there would be a mapping of all sonar metrics and their type because several of them are percentages, but I don't really have the need or interest to map all of them.  Would it make more sense like this?

```
if (metricName === 'coverage') {
  badgeData.text[1] = value.toFixed(0) + '%';
  badgeData.colorscheme = coveragePercentageColor(value);
} else if (metricName === 'tech_debt') {
  // colors are based on sonarqube default rating grid and display colors
  // [0,0.1)   ==> A (green)
  // [0.1,0.2) ==> B (yellowgreen)
  // [0.2,0.5) ==> C (yellow)
  // [0.5,1)   ==> D (orange)
  // [1,)      ==> E (red)
  badgeData.text[1] = value + '%';
  if (value > 100) {
    badgeData.colorscheme = 'red';
  } else if (value > 50) {
    badgeData.colorscheme = 'orange';
  } else if (value > 20) {
    badgeData.colorscheme = 'yellow';
  } else if (value > 10) {
    badgeData.colorscheme = 'yellowgreen';
  } else if (value > 0) {
    badgeData.colorscheme = 'green';
  } else {
    badgeData.colorscheme = 'brightgreen';
  }
} else {
  badgeData.text[1] = metric(value);
  badgeData.colorscheme = 'green';
}
```